org.jacoco.core.test/src/org/jacoco/core/test/validation/ResizeInstructionsTest.java

```diff
@@ -48,6 +49,48 @@ public void teardown() {
 		runtime.shutdown();
 	}
 
+	private class Inner {
+	}
+
+	/**
+	 * Test of ASM bug
+	 * <a href="https://gitlab.ow2.org/asm/asm/issues/317792">#317792</a>.
+	 */
+	@Test
+	public void should_not_loose_InnerClasses_attribute() throws Exception {
+		final ClassWriter cw = new ClassWriter(0);
+		final ClassReader cr = new ClassReader(Java9Support.downgradeIfRequired(
+				TargetLoader.getClassDataAsBytes(Inner.class)));
+		cr.accept(new ClassVisitor(InstrSupport.ASM_API_VERSION, cw) {
+			@Override
+			public void visitEnd() {
+				final MethodVisitor mv = cv.visitMethod(0, "m", "()V", null,
+						null);
+				mv.visitCode();
+				addCauseOfResizeInstructions(mv);
+				mv.visitMaxs(2, 1);
+				mv.visitEnd();
+				super.visitEnd();
+			}
+		}, 0);
+		final byte[] bytes = instrumenter.instrument(cw.toByteArray(), "");
+
+		final TargetLoader targetLoader = new TargetLoader();
+		final Class outer = targetLoader.add(ResizeInstructionsTest.class,
```

Comment:
@Godin Raw `Class` types give a compiler warning, `Class<?>` fixes the issue.

Reply:
@marchof fixed